src/schema/objects/formats.yaml

```
@@ -0,0 +1,115 @@
+---
+# This file defines valid patterns for different formats
+# Entity patterns
+label:
+  name: Label
+  description: |
+    Freeform labels without special characters.
+  pattern: '[0-9a-zA-Z]+'
+index:
+  name: Index
+  description: |
+    Non-negative, non-zero integers, optionally prefixed with leading zeros for sortability.
+    An index may not be all zeros.
+  pattern: '[0-9]*[1-9]+[0-9]*'
+# Metadata types
+string:
+  name: String
+  description: |
+    The basic string type (not a specific format).
+    This should allow any free-form string.
+  pattern: '.*'
+integer:
+  name: Integer
+  description: |
+    An integer which may be positive or negative.
+  pattern: '[+-]?\d+'
+number:
+  name: Number
+  description: |
+    A number which may be an integer or float, positive or negative.
+  pattern: '[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)'
```

I would expect scientific notation (e.g., `1e3`, `-2.1E+5`) to be permissible.

```suggestion
  pattern: '[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)([eE][+-]?[0-9]+)?'
```

---

ohhhh good catch!